### PR TITLE
add doc

### PR DESCRIPTION
**Pull request by Jobhdez**

### CODE REVIEW

File: src/typechecker/derive-type.lisp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ;;;
 
 (defun derive-literal-type (value)
+  "Derive the TYPE and generate a TYPED-NODE for literal VALUE."
```
Comment (stylewarning) [inline src/typechecker/derive-type.lisp:8]: This doc isn't entirely correct since two values are returned.

**Review by stylewarning — CHANGES_REQUESTED** [review]

**Review by eliaslfox — APPROVED** [review]

File: src/typechecker/derive-type.lisp [hunk 2]
```diff
@@ -5,6 +5,7 @@
 ;;;
 
 (defun derive-literal-type (value)
+  "Derive the TYPE of the literal value and the TYPE predicate-list."
```
Comment (stylewarning) [inline src/typechecker/derive-type.lisp:8]: I think something like

> Derive the type of a given literal VALUE. Return two values, the type (a TY) and a list of predicates (a TY-PREDICATE-LIST).

**Review by stylewarning — CHANGES_REQUESTED** [review]

File: src/typechecker/derive-type.lisp [hunk 3]
```diff
@@ -5,6 +5,8 @@
 ;;;
 
 (defun derive-literal-type (value)
+  "Derive the type of a given literal VALUE. 
+  Return two values, the type (a TY) and a list of predicates (a TY-PREDICATE-LIST)."
```
Comment (stylewarning) [inline src/typechecker/derive-type.lisp:9]: Sorry to be so picky, but please don't break lines in this way. I would suggest either:

1. Not breaking lines unless you're starting a new paragraph, in which case you use two new lines with no indentation. This is my preferred style, but is an eye sore to many. 

or:

2. Break lines (but still respecting the aforementioned paragraph rule), using `M-q` in Emacs, autowrapping at 80ish columns. This is not my preferred style, but is easier to read in an editor. 

There basically should never be a newline and whitespace.

**Review by stylewarning — CHANGES_REQUESTED** [review]

**Review by stylewarning — APPROVED** [review]

**Comment by stylewarning** [comment]
@Jobhdez Looks good. Squash commits (with a nice short commit message) and I can merge.

**Comment by Jobhdez** [comment]
> @Jobhdez Looks good. Squash commits (with a nice short commit message) and I can merge.

@stylewarning I squashed the commits I believe. Thanks

**Comment by stylewarning** [comment]
@Jobhdez the commits aren't squashed; there are still 4 commits in this PR.

What I do to squash commits is `git rebase -i HEAD~4` (if I want to squash the last 4 commits), `s` three of them and `r` one of them, write a nice message, exit, and `git push -f` back to the branch.

**Comment by Jobhdez** [comment]
> @Jobhdez the commits aren't squashed; there are still 4 commits in this PR.
> 
> What I do to squash commits is `git rebase -i HEAD~4` (if I want to squash the last 4 commits), `s` three of them and `r` one of them, write a nice message, exit, and `git push -f` back to the branch.

@stylewarning oh ok, ill do that in a sec

**Comment by Jobhdez** [comment]
> > @Jobhdez the commits aren't squashed; there are still 4 commits in this PR.
> > What I do to squash commits is `git rebase -i HEAD~4` (if I want to squash the last 4 commits), `s` three of them and `r` one of them, write a nice message, exit, and `git push -f` back to the branch.
> 
> @stylewarning oh ok, ill do that in a sec

@stylewarning there it is I believe

**Comment by stylewarning** [comment]
> > > @Jobhdez the commits aren't squashed; there are still 4 commits in this PR.
> 
> > > What I do to squash commits is `git rebase -i HEAD~4` (if I want to squash the last 4 commits), `s` three of them and `r` one of them, write a nice message, exit, and `git push -f` back to the branch.
> 
> > 
> 
> > @stylewarning oh ok, ill do that in a sec
> 
> 
> 
> @stylewarning there it is I believe

Commits are squashed, but you'll want to give a nice message (we don't want "squash commits" to be the commit message in the permanent history of this repo), and you'll probably want to rebase against master as well, since you've incidentally brought commits in with a merge commit that you squashed in.

**Comment by Jobhdez** [comment]
> > > > @Jobhdez the commits aren't squashed; there are still 4 commits in this PR.
> > 
> > 
> > > > What I do to squash commits is `git rebase -i HEAD~4` (if I want to squash the last 4 commits), `s` three of them and `r` one of them, write a nice message, exit, and `git push -f` back to the branch.
> > 
> > 
> > > 
> > 
> > 
> > > @stylewarning oh ok, ill do that in a sec
> > 
> > 
> > @stylewarning there it is I believe
> 
> Commits are squashed, but you'll want to give a nice message (we don't want "squash commits" to be the commit message in the permanent history of this repo), and you'll probably want to rebase against master as well, since you've incidentally brought commits in with a merge commit that you squashed in.

@stylewarning I think im finally done. it looks fine now

**Pull request merged**